.github/workflows/sts-mock-pull-request.yml

```diff
@@ -0,0 +1,64 @@
+name: Async-credential Pull Request CI
+on:
+  # Triggers the workflow on pull request events for all branches
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+    paths:
+      - "backend-api/**"
+      - ".github/workflows/backend-api-pull-request.yml"
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  ci-checks:
+    name: Run CI checks
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: backend-api
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - name: Setup nodeJS v20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+          cache-dependency-path: backend-api/package-lock.json
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Linting
+        run: npm run lint
+
+      - name: Check formatting using Prettier
+        run: npm run format:check
+
+      # Runs unit, infra and pact tests
+      - name: Run tests 
+        run: npm run test
+
+      - name: "Run SonarCloud Scan"
```

Comment:
I think we need to either:
- comment these two steps out as we don't have a separate sonar project set up for this stack. If we do uncomment, please may we add an extra comment indicating why it's uncommented, e.g. "Disabling until a separate Sonar project has been set up for this stack"